.github/workflows/run-profiling.yaml

```diff
@@ -24,6 +24,25 @@ jobs:
           UNIQUE_ID=${GITHUB_EVENT_NAME}_${GITHUB_RUN_NUMBER}_${GITHUB_SHA} >> $GITHUB_ENV
```

Comment:
Realise this is not a change from this PR, but I don't think the `>> $GITHUB_ENV` is necessary here. It looks like you only need this environment variable locally in this run step. The `GITHUB_ENV` variable points to a file that can be used to share variables [across job steps](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable) and as its a path to a file the syntax would be
```
echo "{environment_variable_name}={value}" >> "$GITHUB_ENV"
```
I don't think that assigning to a variable outputs anything to `stdout` so I think the above will currently have no effect (appending an empty string to `$GITHUB_ENV` file).